src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1248,6 +1248,41 @@
    ["create index certnames_inactive_idx on certnames using btree (certname)"
     "  where deactivated is not null or expired is not null"]))
 
+(defn switch-to-deduplicated-packages []
+  (jdbc/do-commands
+   ["create table packages "
+    "  (id bigint PRIMARY KEY, "
+    "   hash bytea, "
+    "   name text not null, "
+    "   provider text not null, "
+    "   version text not null)"]
+
+   "CREATE SEQUENCE package_id_seq CYCLE"
+   "ALTER TABLE packages ALTER COLUMN id SET DEFAULT nextval('package_id_seq')"
+
+   ["ALTER TABLE ONLY packages "
+    "ADD CONSTRAINT package_hash_key UNIQUE (hash)"]
+
+   ["insert into packages (name, provider, version) "
+    "select distinct name, provider, version "
+    "from package_inventory"]
+
+   ["create table certname_packages"
+    "  (certname_id bigint not null,"
+    "   package_id integer not null,"
```

Comment:
I think we will want a bigint for package_id here. right now it won't actually support the same range of ids as the packages table